src/appState.ts

```diff
@@ -62,30 +62,86 @@ export const getDefaultAppState = (): AppState => {
   };
 };
 
-export const clearAppStateForLocalStorage = (appState: AppState) => {
-  const {
-    draggingElement,
-    resizingElement,
-    multiElement,
-    editingElement,
-    selectionElement,
-    isResizing,
-    isRotating,
-    collaborators,
-    isCollaborating,
-    isLoading,
-    errorMessage,
-    showShortcutsDialog,
-    editingLinearElement,
-    isLibraryOpen,
-    ...exportedState
-  } = appState;
-  return exportedState;
+const APP_STATE_STORAGE_CONF = (<
+  Values extends { browser: boolean; export: boolean },
+  T extends Record<keyof AppState, Values>
+>(
+  config: { [K in keyof T]: K extends keyof AppState ? T[K] : never },
+) => config)({
+  isLoading: { browser: false, export: false },
+  errorMessage: { browser: false, export: false },
+  draggingElement: { browser: false, export: false },
+  resizingElement: { browser: false, export: false },
+  multiElement: { browser: false, export: false },
+  editingElement: { browser: false, export: false },
+  editingLinearElement: { browser: false, export: false },
+  isCollaborating: { browser: false, export: false },
+  isResizing: { browser: false, export: false },
+  isRotating: { browser: false, export: false },
+  selectionElement: { browser: false, export: false },
+  collaborators: { browser: false, export: false },
+  showShortcutsDialog: { browser: false, export: false },
+  width: { browser: false, export: false },
+  height: { browser: false, export: false },
+  elementType: { browser: true, export: false },
+  elementLocked: { browser: true, export: false },
+  exportBackground: { browser: true, export: false },
+  shouldAddWatermark: { browser: true, export: false },
+  currentItemStrokeColor: { browser: true, export: false },
+  currentItemBackgroundColor: { browser: true, export: false },
+  currentItemFillStyle: { browser: true, export: false },
+  currentItemStrokeWidth: { browser: true, export: false },
+  currentItemStrokeStyle: { browser: true, export: false },
+  currentItemRoughness: { browser: true, export: false },
+  currentItemOpacity: { browser: true, export: false },
```

Comment:
shouldn't this be just for current item? right now it gets affected for all future items. This is a separate discussion though

Reply:
The naming is weird, but the idea is to retain the props for following shapes you draw. That said, we may wanna investigate splitting this in two so that when you change a prop of an existing element, it doesn't propagate back to appState. Maybe be unnecessary complexity, dunno.